src/loadRc.js

```diff
@@ -32,72 +33,61 @@ module.exports = function loadRc(
   }
 
   function parseExtensionlessRcFile(content: ?string): ?cosmiconfig$Result {
-    if (!content) return null;
-    const pasedConfig = options.rcStrictJson
+    const isEmpty = content === '';
```

Comment:
Really all three methods are doing the exact same thing just parsing differently.  Instead of calling a helper you could utilize a factory e.g.

*not sure 'factory' is the correct term here, I'm no academic ha*

```js
// createParseFile.js
module.exports = function createParseFile(filepath, parse, ignoreEmpty) {
  return function parseFile(content) {
    const isEmpty = content === '';
    if (content == null || (isEmpty && ignoreEmpty)) return null;

    if (isEmpty) {
      return {
        config: undefined,
        filepath,
        isEmpty,
      };
    }
    
    return {
      config: parse(parsedConfig),
      filepath,
    };
  }
}
```

and consume it like

```js
// loadRc.js
const parseYaml = (content, filepath) => yaml.safeLoad(content, { filename: filepath });

const parse = options.rcStrictJson
  ? parseJson
  : parseYaml

const parseExtensionlessRcFile = createParseFile(filepath, parse, ignoreEmpty)

// ...
// inside loadExtension

const fpath = `${filepath}.${extn}`
const parseRcFile = createParseFile(fpath, parse, ignoreEmpty)

// then replace
return funcRunner(readRcFile(extn), [
// with
return funcRunner(readRcFile(extn), [parseRcFile])
```

```js
// loadJs.js
const parseJsFile = createParseFile(filepath, requireFromString, ignoreEmpty)
```

Reply:
My main intention is to avoid code repetition and what you suggested looks good to me.